lib/archethic/oracle_chain.ex

```diff
@@ -186,4 +190,82 @@ defmodule Archethic.OracleChain do
       |> DateTime.from_naive!("Etc/UTC")
     end
   end
+
+  @doc """
+    Methods Serves to resolve Situation
+    time  : event
+    12:00 : auth node rejoins
+    12:01:40 : completes sync ,recieves oracle summary to be verified by this auth node
+    12:01:55 : will result failure as last oracle_gen_addr not there
+  """
+  @spec update_last_summ_gen_addr :: :ok
+  def update_last_summ_gen_addr() do
```

Comment:
Instead of calculating the last summary_date, you can just use `OracleChain.previous_summary_date()`